test/testing_utils.dart

```diff
@@ -0,0 +1,224 @@
+  return driver;
+}
+
+Future navigateToPage(WebDriver driver, String slug) async {
+  await driver.get("https://www.gyerunkanyukam.hu/" + slug);
+}
+
+Future<String> signUpNewRandomUser(WebDriver driver, String pw) async {
+  var newRandomUserEmail;
+  var loginBtn = await driver
+      .findElement(const By.cssSelector('div.menu-item.menu-item--login > a'));
+  await loginBtn.click();
+  var signUpBtn = await driver
+      .findElement(const By.xpath('//*//tab-button[2]/material-ripple'));
+  await signUpBtn.click();
+  var signUpEmail = await driver.findElement(const By.xpath(
+      '*//label[contains(@class,"input-container")]/input[contains(@type,"email")]'));
+  newRandomUserEmail = generateRandomEmail();
+  await signUpEmail.sendKeys(newRandomUserEmail);
+  var signUpPW = await driver.findElement(const By.xpath(
+      '*//label[contains(@class,"input-container")]/input[contains(@type,"password")]'));
+  await signUpPW.sendKeys(pw);
+  var termsRadioBtn = await driver.findElement(const By.xpath(
+      "*//material-checkbox/div[contains(@class, 'icon-container')]/material-ripple"));
+  await termsRadioBtn.click();
+  var submitBtn =
+      await driver.findElement(const By.xpath('*//input[@type="submit"]'));
+  await submitBtn.click();
+  await waitForAngular(driver);
+  await sleep5();
+  return newRandomUserEmail;
+}
+
+String generateRandomEmail() {
+  return random.randomAlphaNumeric(10) + "@email.com";
+}
+
+Future loginUserWithCredentials(
+    String username, String pw, WebDriver driver, bool removeOverlay) async {
+  bool loggedIn = await isLoggedIn(driver);
+  if (loggedIn) {
+    print("loginUserWithCredentials: Already logged in, logging out");
+    await logoutUser(driver);
+  }
+  print("loginUserWithCredentials: Logging in as " + username);
+  await navigateToPage(driver, "login");
+  await waitForAngular(driver);
+  await sleep10();
+  var loginTabBtn = await driver
+      .findElement(const By.xpath('*//tab-button[1]/material-ripple'));
+  await loginTabBtn.click();
+  var loginEmail = await driver.findElement(const By.xpath(
+      '*//label[contains(@class,"input-container")]/input[contains(@type,"email")]'));
+  await loginEmail.sendKeys(username);
+  var loginPW = await driver.findElement(const By.xpath(
+      '*//label[contains(@class,"input-container")]/input[contains(@type,"password")]'));
+  await loginPW.sendKeys(pw);
+  var submitBtn =
+      await driver.findElement(const By.xpath('*//input[@type="submit"]'));
+  await submitBtn.click();
+  await waitForAngular(driver);
+  if (removeOverlay) {
+    await sleep10();
+    await dismissOverlay(driver);
+  }
+  print(
+      "loginUserWithCredentials: Logged in as user " + username + " pw: " + pw);
+}
+
+Future dismissOverlay(WebDriver driver) async {
+  var continueBtn;
+  try {
+    await waitForAngular(driver);
+    continueBtn = await driver.findElement(const By.cssSelector(
+        'product-module > cutout-shadow-overlay > div > div.cutout'));
```

Comment:
If you implement this as a page object, you can use the `@optional` annotation on the `PageLoaderElement`, indicating that the given element may or may not be present in the DOM.